src/app/shared/example-viewer/example-viewer.spec.ts

```diff
@@ -48,12 +48,57 @@ describe('ExampleViewer', () => {
 
   it('should toggle between the 3 views', async(() => {
     fixture.detectChanges();
-    component.view = 'compact';
-    expect(component.view).toBe('compact');
+    // need to specify a file because toggling from snippet to full changes the tabs to match
+    component.file = 'file.html';
+    component.view = 'snippet';
+    expect(component.view).toBe('snippet');
     component.toggleCompactView();
     expect(component.view).toBe('full');
     component.toggleSourceView();
-    expect(component.view).toBe('collapsed');
+    expect(component.view).toBe('demo');
+  }));
+
+  it('should change to correct tab', async(() => {
+    fixture.detectChanges();
+    component.file = 'file.html';
+    component.selectCorrectTab();
+    expect(component.selectedTab).toBe(0);
```

Comment:
Hmm, can we update this assert to be a bit more expressive. We actually don't actually care about the index number. What we care about is that when the file is `html`, we selected the `html` tab, regardless of what index that actually is. Can you use the tab harnesses to get the selected tab label? https://github.com/angular/components/tree/master/src/material/tabs/testing